src/main/kotlin/com/moshbit/katerbase/MongoDatabaseDefinition.kt

```diff
@@ -12,7 +13,28 @@ class MongoDatabaseDefinition {
     val collectionName: String,
     val collectionSizeCap: Long?
   ) {
-    class Index(val index: Bson, val partialIndex: Array<FilterPair>?, val indexOptions: (IndexOptions.() -> Unit)?)
+    class Index(val index: Bson, val partialIndex: Array<FilterPair>?, val indexOptions: (IndexOptions.() -> Unit)?) {
+      init {
+        partialIndex?.forEach { filterPair ->
+          (filterPair.value as? Document)?.entries?.forEach { (operator: String, value: Any) ->
+            // Not all operators are allowed for partial indexes, see https://docs.mongodb.com/manual/core/index-partial/#create-a-partial-index
+            when {
+              operator.removePrefix("$") in allowedPartialIndexQueryOperators -> Unit
+              operator == "\$exists" && value == true -> {
```

Comment:
for consistency: please just add a `$` to all values in allowedPartialIndexQueryOperators. then we don't need to check once for removePrefix(`$`) variant and once for withPrefix (`"\$exists"`)